examples/webgl_lights_spherical_harmonics.html

```diff
@@ -0,0 +1,265 @@
+				camera.position.set( 0, 0, 250 );
+
+				scene = new THREE.Scene();
+				
+				// LIGHTS
+
+				// Spherical Harmonic Values retrieved from textures/cube/skybox
+				// via Unity3D and this script: https://gist.github.com/atteneder/d4fe33368afaf176413ec09d14ef41db
+				var sh3 = new THREE.SphericalHarmonics3().set(
+		    		new THREE.Color(0.3996281,0.3411802,0.2419128),
+					new THREE.Color(0.142093,0.09635267,0.01494713),
+					new THREE.Color(0.08023436,0.07530768,0.03336681),
+					new THREE.Color(0.07269116,0.07394479,0.04278772),
+					new THREE.Color(0.05431392,0.05045794,0.0205737),
+					new THREE.Color(0.07121608,0.06678943,0.03363363),
+					new THREE.Color(0.008722264,0.005476702,0.0006165598),
+					new THREE.Color(0.01847383,0.03232491,0.02740111),
+					new THREE.Color(0.02490644,0.0140671,-0.002444069)
+				);
+
+				//
+				// Further example values to try:
+				// Debevec's light probe gallery examples.  Source: http://www.yaldex.com/open-gl/ch12lev1sec3.html
+				//
+
+				THREE.SphericalHarmonics3.EucalyptusGroove = new THREE.SphericalHarmonics3().set(
+				    new THREE.Color( 0.3783264,  0.4260425,  0.4504587),
+				    new THREE.Color( 0.2887813,  0.3586803,  0.4147053),
+				    new THREE.Color( 0.0379030,  0.0295216,  0.0098567),
+				    new THREE.Color(-0.1033028, -0.1031690, -0.0884924),
+				    new THREE.Color(-0.0621750, -0.0554432, -0.0396779),
+				    new THREE.Color( 0.0077820, -0.0148312, -0.0471301),
+				    new THREE.Color(-0.0935561, -0.1254260, -0.1525629),
+				    new THREE.Color(-0.0572703, -0.0502192, -0.0363410),
+				    new THREE.Color( 0.0203348, -0.0044201, -0.0452180)
+				);
+
+				THREE.SphericalHarmonics3.FunstoneBeachSunset = new THREE.SphericalHarmonics3().set(
+				    new THREE.Color( 0.6841148,  0.6929004,  0.7069543),
+				    new THREE.Color( 0.3173355,  0.3694407,  0.4406839),
+				    new THREE.Color(-0.1747193, -0.1737154, -0.1657420),
+				    new THREE.Color(-0.4496467, -0.4155184, -0.3416573),
+				    new THREE.Color(-0.1690202, -0.1703022, -0.1525870),
+				    new THREE.Color(-0.0837808, -0.0940454, -0.1027518),
+				    new THREE.Color(-0.0319670, -0.0214051, -0.0147691),
+				    new THREE.Color( 0.1641816,  0.1377558,  0.1010403),
+				    new THREE.Color( 0.3697189,  0.3097930,  0.2029923)
+				);
+
+				THREE.SphericalHarmonics3.GalileoTomb = new THREE.SphericalHarmonics3().set(
+				    new THREE.Color( 1.0351604,  0.7603549,  0.7074635),
+				    new THREE.Color( 0.4442150,  0.3430402,  0.3403777),
+				    new THREE.Color(-0.2247797, -0.1828517, -0.1705181),
+				    new THREE.Color( 0.7110400,  0.5423169,  0.5587956),
+				    new THREE.Color( 0.6430452,  0.4971454,  0.5156357),
+				    new THREE.Color(-0.1150112, -0.0936603, -0.0839287),
+				    new THREE.Color(-0.3742487, -0.2755962, -0.2875017),
+				    new THREE.Color(-0.1694954, -0.1343096, -0.1335315),
+				    new THREE.Color( 0.5515260,  0.4222179,  0.4162488)
```

Comment:
This is not the right design. These are _coefficients_, not colors. Colors can't be negative.

Reply:
yeah, I guess you are right.  I guess we should replace them with vectors.

Reply:
You have to admit that in the SphericalHarmonic3 class the Color class actually works quite well and is a convenient way to represent the coefficients, both for retrieval and accumulation: https://github.com/mrdoob/three.js/pull/13115/files#diff-b4c03c6f0f7c31a8942572bd76e5068d

Reply:
Thinking about it, I completely disagreed.  They are coefficients but they are also colors.  They are the colors of the various SH bulbs.  Yes, sometimes they are negative but that is allowed when they are color coefficients.

Look on page 13 - basically the color coefficiences are the intensities of these individual bulbs added up -- in our case just the first 3 layers of 9 bulbs: http://silviojemma.com/public/papers/lighting/spherical-harmonic-lighting.pdf

So I think these should say colors even though they are color efficients.

Reply:
>They are coefficients but they are also colors.

@bhouston What color is ( - 1, 0, 0 )?

Reply:
I guess I like the beauty of this code here: https://github.com/mrdoob/three.js/pull/13115/files#diff-b4c03c6f0f7c31a8942572bd76e5068dR100

Notice it just sums up the coefficiencies with their weights and the end result is a color.

These are color coefficiencies, it is just that when you decompose it, you may end up with some negative colors that are balanced when summing them together.  Call it is a virtual color that is never realized by itself, but only in combination with the other SH color lobes?

Reply:
This code is really nice too: https://github.com/mrdoob/three.js/pull/13115/files#diff-b4c03c6f0f7c31a8942572bd76e5068dR69

To convert it to a Vector3 seems like it just introduces more conversions and inconvenience.  So yet, I am arguing from style back to this being a "color" but I like the way the code is.  Also it sort of has to be a color because I never do anything but scale and sum them, there is really no true conversion taking place.

Also more proof that technically these are colors, band 0 of the standard SH is just a single color and it it just scaled by the coefficient 1.  That seems like a color to me.

Reply:
According to wikipedia: https://en.wikipedia.org/wiki/Spherical_harmonics (see the main picture), the coefficiencies can be negative based on convention as they are defining the magnitude of periodic like functions.  Just like the frequency amplitudes resulting from an Fourier transform  of a real signal can be negative.  It doesn't mean there are negative frequencies (again, you could ask, does negative frequencies exist) but it is related to the phase of the signal that allows it to be negative, because it is a periodic process.

Same thing.

Reply:
Sorry, I have hard time stomaching negative colors. We have the method`Vector3.addScaledVector( v, s )`if you insist on that parameterization.

Spherical Harmonics are univariate, and unit-less. An `SHLight` can be parameterized by 9 _coefficients_ for each of 3 channels. Given the coefficients, the radiance and irradiance is then _computed_.

Reply:
"Spherical Harmonics are univariate, and unit-less."

They are not unit less, otherwise how would we produce a color as a result?  Working backwards from the typed result, one of the two parts of the products must be unitless, either the variable coefficient or the fixed weights for the various harmonics.  I am pretty sure the fixed weights are unitless, and because the result is a typed result (in our case color), then the coefficiencies must be typed.

This can seen because SH can represent many different types of qualities and in each case, the coefficiencies takes on the same properties of those types (intensity, color, etc.)

But it isn't important to me, so someone can swithc it to vector, it is just a type conversion.

Reply:
Color is unit-less. It is just a mask, and hence must be non-negative. It is the intensity parameter that has the units.

That is how I think about it anyway.

And, yes, the implementation can be changed later if we want.